tests/test_utils.py

```diff
@@ -0,0 +1,83 @@
+
+
+class TestBuildTerraformCommand(TestCase):
+    def test_happy_path(self) -> None:
+        expected_terraform_command = """\
+#!/bin/bash
+set -ex
+TF_WORKSPACE=test terraform -chdir=/some/module/path init -upgrade -input=false
+TF_WORKSPACE=test TF_LOG=INFO terraform -chdir=/some/module/path apply --auto-approve -var-file=test.tfvars
+        """
+        self.assertEqual(
+            dedent(expected_terraform_command),
+            build_terraform_command(
+                terraform_workspace='test',
+                terraform_vars_file='test.tfvars',
+                terraform_vars={},
+                module_path='/some/module/path',
+                log_level=TerraformLogLevel.INFO,
+            ),
+        )
+
+    def test_single_terraform_vars(self) -> None:
+        expected_terraform_command = """\
+#!/bin/bash
+set -ex
+TF_WORKSPACE=test terraform -chdir=/some/module/path init -upgrade -input=false
+TF_WORKSPACE=test TF_LOG=INFO terraform -chdir=/some/module/path apply --auto-approve -var-file=test.tfvars -var='hello=world'
+        """
+        self.assertEqual(
+            dedent(expected_terraform_command),
+            build_terraform_command(
+                terraform_workspace='test',
+                terraform_vars_file='test.tfvars',
+                terraform_vars={'hello': 'world'},
+                module_path='/some/module/path',
+                log_level=TerraformLogLevel.INFO,
+            ),
+        )
+
+    def test_multiple_terraform_vars(self) -> None:
+        expected_terraform_command = """\
+#!/bin/bash
+set -ex
+TF_WORKSPACE=test terraform -chdir=/some/module/path init -upgrade -input=false
+TF_WORKSPACE=test TF_LOG=INFO terraform -chdir=/some/module/path apply --auto-approve -var-file=test.tfvars -var='hello=world' -var='wee=womp'
+        """
+        self.assertEqual(
+            dedent(expected_terraform_command),
+            build_terraform_command(
+                terraform_workspace='test',
+                terraform_vars_file='test.tfvars',
+                terraform_vars={'hello': 'world', 'wee': 'womp'},
+                module_path='/some/module/path',
+                log_level=TerraformLogLevel.INFO,
+            ),
+        )
+
+
+class TestTerraformDeploy(TestCase):
+    @patch('tests.utils.os.chmod')
+    @patch('tests.utils.os.execvp')
+    def test_happy_path(self, mock_os_execvp: Mock, mock_os_chmod: Mock) -> None:
+        terraform_deploy(
+            terraform_workspace='test',
+            terraform_vars_file='test.tfvars',
+            terraform_vars={},
+            module_path='/some/module/path',
+            log_level=TerraformLogLevel.INFO,
+        )
+        mock_os_execvp.assert_called_once()
+        mock_os_chmod.assert_called_once()
+
+        # Ensure the script written to disk is executable
+        _, perm_bits = mock_os_chmod.mock_calls[0][1]
+        filemode: str = stat.filemode(perm_bits)
+        self.assertEqual('-rwx------', filemode)
```

Comment:
how does this work? didn't you mock out the chmod?

Reply:
This just asserts we call chmod with the correct file mode. It' mocked